src/charm.py

```diff
@@ -187,22 +255,21 @@ def configure_workload_container(self, event) -> None:
                 logger.info(msg)
                 self.unit.status = MaintenanceStatus(msg)
                 container.add_layer(CONTAINER_NAME, pebble_config, combine=True)
+                container.pebble.replan_services()
+        else:
+            self.unit.status = WaitingStatus("waiting for Pebble to start")
+            event.defer()
+            return
 
-                service = container.get_service(CONTAINER_NAME)
-                if service.is_running():
-                    msg = "Stopping content-cache"
-                    logger.info(msg)
-                    self.unit.status = MaintenanceStatus(msg)
-                    container.stop(CONTAINER_NAME)
-
-                msg = "Starting content-cache"
-                logger.info(msg)
-                self.unit.status = MaintenanceStatus(msg)
-                container.start(CONTAINER_NAME)
-        except ConnectionError:
-            msg = "Pebble is not ready, deferring event"
+        exporter_container = self.unit.get_container(EXPORTER_CONTAINER_NAME)
+        if exporter_container.can_connect():
+            msg = "Updating exporter pebble layer config"
             logger.info(msg)
-            self.unit.status = WaitingStatus(msg)
+            self.unit.status = MaintenanceStatus(msg)
+            exporter_container.add_layer(EXPORTER_CONTAINER_NAME, exporter_config, combine=True)
+            exporter_container.pebble.replan_services()
+        else:
+            self.unit.status = WaitingStatus("waiting for Pebble to start")
```

Comment:
Same comment as above about sentence casing, and including which container we're waiting to start